react/features/whiteboard/index.tsx

```diff
@@ -0,0 +1,144 @@
+/* eslint-disable import/order */
+/* eslint-disable lines-around-comment */
+
+/* eslint-ignore */
+// @ts-ignore
+import { ExcalidrawApp } from '@jitsi/excalidraw';
+
+import React, { useCallback, useEffect, useRef } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import clsx from 'clsx';
+
+// @ts-ignore
+import Filmstrip from '../../../modules/UI/videolayout/Filmstrip';
+// @ts-ignore
+import { getVerticalViewMaxWidth } from '../filmstrip/functions.web';
+// @ts-ignore
+import { getToolboxHeight } from '../toolbox/functions.web';
+// @ts-ignore
+import { getCollabLink, getCollabServerUrl, getUsernameStatus, isWhiteboardVisible } from './functions';
+// @ts-ignore
+import { shouldDisplayTileView } from '../video-layout/functions.any';
+
+import { setUsernameStatus } from './actions';
+// @ts-ignore
+import { getLocalParticipant } from '../base/participants';
+import { USERNAME_UPDATE_STATUS } from './constants';
+
+const HEIGHT_OFFSET = 80;
+
+declare const interfaceConfig: any;
+
+interface IDimensions {
+
+    /* The height of the component. */
+    height: string;
+
+    /* The width of the component. */
+    width: string;
+}
+
+/**
+ * The Whiteboard component.
+ *
+ * @returns {JSX.Element} - The React component.
+ */
+const Whiteboard: () => JSX.Element = () => {
+    const excalidrawRef = useRef<any>(null);
+    const collabAPIRef = useRef<any>(null);
+    const dispatch = useDispatch();
+
+    const shouldDisplay = useSelector(isWhiteboardVisible);
+    const isInTileView = useSelector(shouldDisplayTileView);
+    const { clientHeight, clientWidth } = useSelector((state: any) => state['features/base/responsive-ui']);
+    const { visible: filmstripVisible, isResizing } = useSelector((state: any) => state['features/filmstrip']);
+    const filmstripWidth: number = useSelector(getVerticalViewMaxWidth);
+    const collabLink = useSelector(getCollabLink);
+    const collabServerUrl = useSelector(getCollabServerUrl);
+    // @ts-ignore
+    const localParticipantName: string = useSelector(getLocalParticipant).name;
+    const usernameStatus = useSelector(getUsernameStatus);
+
+    useEffect(() => {
+        if (!collabAPIRef.current || usernameStatus !== 'NEW') {
+            return;
+        }
+
+        collabAPIRef.current.setUsername(localParticipantName);
+        dispatch(setUsernameStatus(USERNAME_UPDATE_STATUS.UPDATED));
+    }, [ usernameStatus ]);
```

Comment:
@saghul here's where I use the usernameStatus. Since you can update your display name during meetings, we're making sure that the whiteboard also gets the message and displays the latest name next to your cursor. WDYT?

Reply:
I wanted to do this somewhere else, in a middleware preferably, but I can't think of a way to access the collaboration API outside the react component.

Reply:
But can's the actual display name from a prop and you just list that as a dependency in the array? If it changes your effect will run and you call setUsername again no? Am I missing something?

Reply:
Oh you mean to add the localParticipantName as a dependency and drop the other one? Sounds actually pretty neat this way. Thanks! I went down the rabbit whole on this one and missed the big picture

Reply:
👍